### ISSUE TITLE
Cannot override methods when extending WebGLRenderer

### ISSUE THREAD

**Issue by marcofugaro** [mrdoob/three.js#25598]
### Description

In my case, I have a `CustomRenderer` class that extends `WebGLRenderer`. I want to dispose of other things in the `dispose()` call, so I created an override method that disposes of my resources and also calls `super.dispose()`.

However, in WebGLRenderer, the method doesn't ever get called, and instead the `WebGLRenderer.dispose()` is called directly.

```js
class CustomRenderer extends WebGLRenderer {
  dispose() {
    super.dispose()
    // the code here is never run..
  }
}

const renderer = new CustomRenderer()
renderer.dispose()
```

This is because the `WebGLRenderer.dispose()`, is not a class method per se.

### Reproduction steps

1. Create a class that extends `WebGLRenderer`
2. Create a method that overrides a `WebGLRenderer` method
3. Instance the class and call the method

### Code

```js
import { WebGLRenderer } from 'three';

class CustomRenderer extends WebGLRenderer {
  dispose() {
    super.dispose()
    
    // custom code goes here...
    console.log('1. I should be logged to the console')
  }
}

const renderer = new CustomRenderer()
renderer.dispose()

console.log('2. The log n. 1 should be present above this ^')
```

### Live example

* [jsfiddle](https://jsfiddle.net/8o9hqef7/7/)

### Version

r150

**Comment by marcofugaro** [comment]
Looking at the [WebGLRenderer](https://github.com/mrdoob/three.js/blob/dev/src/renderers/WebGLRenderer.js) class more deeply, I believe this can be achieved when we can use private class properties.

**Comment by agargaro** [comment]
You can override in this way but of course is not clean 

```
class CustomRenderer extends WebGLRenderer {

  constructor() {
  	super();
	const disposeBase = this.dispose;
	this.dispose = () => {
        disposeBase();
		// custom code goes here...
		console.log('1. I should be logged to the console')
    };
  }
}
```

**Comment by marcofugaro** [comment]
Yup, I ended up calling the `dispose()` method a different way.

**Comment by epreston** [comment]
Grab a reference to the original dispose method, replace dispose method on original, use the reference in the body of your replacement to do an effective call to super.

**Comment by mrdoob** [comment]
@marcofugaro Did #25599 address this?

**Comment by Mugen87** [comment]
We can't use private class fields since we want to stick to ECMA2018. So if `WebGLRenderer` should really be refactored, private properties have to use the `_` prefix to mark them as private.

**Comment by marcofugaro** [comment]
Yup, should we refactor using `_` now or should we wait until we can use a new ECMA version in your opinion?

**Comment by Mugen87** [comment]
> Yup, should we refactor using _ now or should we wait until we can use a new ECMA version in your opinion?

We have lately decided to stick to ECMAScript 2018 (see #25185)and I would prefer to not revisit this topic in the next time.

**Comment by marcofugaro** [comment]
@Mugen87 you misread my question, I asked if we should use `_` like you did in your PR or wait until we use a new ECMA version.

**Comment by marcofugaro** [comment]
@Mugen87 you misread my question. I asked if we should use `_` like you did in your PR (we can do this with ECMA2018), or wait until we can use a new ECMA version and use private class fields.

**Comment by Mugen87** [comment]
Sorry, you're right! We can use the `_` approach now and later replace it with private class properties. So we don't have to wait.

I'm just hesitating since refactoring big classes always introduces the risk of breaking something. But I guess sooner or later the mentioned inheritance issue with the renderer would come up again. My hope was to avoid this as long as possible^^.
